include/openenclave/internal/switchless.h

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _SWITCHLESS_H_
+#define _SWITCHLESS_H_
+
+#include <openenclave/bits/result.h>
+#include <openenclave/corelibc/stdint.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/internal/lockless_queue.h>
+
+#ifndef EXTERN
+#ifdef __cplusplus
+#define EXTERN extern "C"
```

Comment:
Why do we need a special case for C++?
Try to reuse existing definitions in bits/defs.h